arch/arm/src/armv7-m/arm_doirq.c

```diff
@@ -56,7 +56,13 @@ void exception_direct(void)
 
 uint32_t *arm_doirq(int irq, uint32_t *regs)
 {
-  struct tcb_s *tcb = this_task();
+  struct tcb_s **running_task = &g_running_tasks[this_cpu()];
+  FAR struct tcb_s *tcb;
+
+  if (*running_task != NULL)
+    {
+      (*running_task)->xcp.regs = regs;
```

Comment:
ditto

Reply:
We can't change it to this way. 
(*running_task)->xcp.regs  to *running_task->xcp.regs

error: 'running_task' is a pointer to pointer; did you mean to dereference it before applying '->' to it?
  167 |       *running_task->xcp.regs = regs;
      |                    ^~
      |        (*          )